src/fontra/views/editor/scene-controller.js

```diff
@@ -1333,6 +1371,73 @@ function getSelectedContours(path, pointSelection) {
   return [...selectedContours];
 }
 
+function getSelectedOpenContours(
+  path,
+  pointSelection,
+  ignoreSinglePointContour = true
+) {
+  if (!path || !pointSelection) {
+    return [];
+  }
+  const selectedContours = new Set();
+  for (const contourIndex of getSelectedContours(path, pointSelection)) {
+    if (path.contourInfo[contourIndex].isClosed) {
+      // skip if contour is closed already
+      continue;
+    }
+    const contour = path.getContour(contourIndex);
+    const numOnCurvePoints = contour.pointTypes.reduce(
+      (acc, pointType) =>
+        acc +
+        ((pointType & VarPackedPath.POINT_TYPE_MASK) === VarPackedPath.ON_CURVE
+          ? 1
+          : 0),
+      0
+    );
+    if (numOnCurvePoints === 1 && ignoreSinglePointContour) {
+      // skip single point contour
+      continue;
+    }
+    selectedContours.add(contourIndex);
+  }
+
+  return [...selectedContours];
+}
+
+function closeContourEnsureCubicOffCurves(path, contourIndex) {
+  const contourInfo = path.contourInfo[contourIndex];
```

Comment:
You can use `path.getContourPoint(contourIndex, contourPointIndex)` to get points more conveniently. `contourPointIndex` can be negative to get points from the end.

There is also `path.getNumPointsOfContour(contourIndex)` to more conveniently get number of points of a contour.

Your "situation" variable can be named better. We have two situations:
- dangling off-curve at the end
- dangling off-curve at the start
Perhaps "offCurveAtEnd" or "offCurveAtStart" is a better name?

Reply:
Thank you very much. done.